packages/service-library/src/servicelib/aiohttp/application_setup.py

```diff
@@ -137,38 +190,52 @@ def _wrapper(app: web.Application, *args, **kargs) -> bool:
 
             if category == ModuleCategory.ADDON:
                 # NOTE: ONLY addons can be enabled/disabled
-                # TODO: sometimes section is optional, check in config schema
-                cfg = app[APP_CONFIG_KEY]
-
-                try:
-                    is_enabled = _is_app_module_enabled(
-                        cfg, config_enabled.split("."), section
-                    )
-                except KeyError as ee:
-                    raise ApplicationSetupError(
-                        f"Cannot find required option '{config_enabled}' in app config's section '{ee}'"
-                    ) from ee
 
+                # TODO: cfg will be fully replaced by app_settings section below
+                cfg = app[APP_CONFIG_KEY]
+                is_enabled = _is_addon_enabled_from_config(cfg, config_enabled, section)
                 if not is_enabled:
                     logger.info(
                         "Skipping '%s' setup. Explicitly disabled in config",
                         module_name,
                     )
                     return False
 
+                # NOTE: if not disabled by config, it can be disabled by settings (tmp while legacy maintained)
+                app_settings, module_settings_name = _get_app_settings_and_field_name(
+                    app,
+                    module_name,
+                    settings_name,
+                    setup_func.__name__,
+                    logger,
+                )
+
+                if (
+                    app_settings
+                    and module_settings_name
+                    and not app_settings.is_enabled(module_settings_name)
+                ):
+                    logger.info(
+                        "Skipping setup %s. %s disabled in settings",
+                        f"{module_name=}",
+                        f"{module_settings_name=}",
+                    )
+                    return False
+
             if depends:
-                uninitialized = [
-                    dep for dep in depends if dep not in app[APP_SETUP_KEY]
-                ]
+                # TODO: no need to enforce. Use to deduce order instead.
```

Comment:
I am not sure I get the meaning of that comment. you raise an exception if a dependency is missing right? is that not enforcing?